core/http/server-spring/src/main/java/org/eclipse/rdf4j/http/server/repository/transaction/Transaction.java

```diff
@@ -117,14 +155,10 @@ UUID getID() {
 	void begin(IsolationLevel level)
 		throws InterruptedException, ExecutionException
 	{
-		Future<Boolean> result = executor.submit(() -> {
+		Future<Boolean> result = submit(() -> {
```

Comment:
Regardless of anything else, having a centralized `sumbit` method that can take care of all the book-keeping makes sense. 👍